client/apps.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015-2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+type AppInfo struct {
+	Snap         string `json:"snap,omitempty"`
+	Name         string `json:"name"`
+	DesktopFile  string `json:"desktop-file,omitempty"`
+	*ServiceInfo `json:",omitempty"`
+}
+
+// IsService returns true if the application is a background daemon.
+func (a *AppInfo) IsService() bool {
+	if a == nil {
+		return false
+	}
+	if a.ServiceInfo == nil {
+		return false
+	}
+	if a.ServiceInfo.Daemon == "" {
+		return false
+	}
+
+	return true
+}
+
+type ServiceInfo struct {
+	Daemon          string `json:"daemon"`
+	ServiceFileName string `json:"service-file-name"`
+	Enabled         bool   `json:"enabled"`
+	Active          bool   `json:"active"`
```

Comment:
We discussed this in our call about that PR a few weeks ago, but happy to go over it again: we're already over that bridge of flattening the terminology, as all of these are actually apps that have commands, per the snap.yaml and snapcraft.yaml files. Each one of those apps have commands, including the ones with the daemon field.

These apps also _become_ commands in the user system, which is why we have them listed under "commands" in the info output. Those things which today are put under systemd for activation should _also_ be commands, in fact, and the lack of that is something that frustrates me and most of the users I raise the point with. And whether there's agreement about doing that or not, the underlying point remains: it's very plausible, which highlights the fact "commands" is a thing on itself, rather than being a class of applications we have.